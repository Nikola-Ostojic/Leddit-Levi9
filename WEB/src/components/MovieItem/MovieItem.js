import React from "react";
import { Card, CardContent } from "@material-ui/core";

import styles from "./MovieItem.module.css";

const MovieItem = ({ movie, onClick }) => (
  <Card
    className={styles.container}
    raised
    onClick={() => onClick(`/movieDetails/${movie.id}`)}
  >
    <CardContent>{movie.name}</CardContent>
  </Card>
);

export default MovieItem;
