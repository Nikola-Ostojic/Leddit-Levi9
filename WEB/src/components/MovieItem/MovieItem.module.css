.container {
    margin-top: 10px;
    box-sizing: border-box;
}