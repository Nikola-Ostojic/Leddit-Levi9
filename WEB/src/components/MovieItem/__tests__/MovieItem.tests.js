import React from "react";
import { shallow } from "enzyme";

import { Card } from "@material-ui/core";

import MovieItem from "../MovieItem";

describe("MovieItem component tests", () => {
  const mockedProps = {
    movie: {
      name: "Test"
    },
    onClick: jest.fn()
  };
  it("Should match snapshot", () => {
    const wrapper = shallow(<MovieItem {...mockedProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("Should call on click when item is clicked", () => {
    const wrapper = shallow(<MovieItem {...mockedProps} />);

    wrapper.find(Card).simulate("click");

    expect(mockedProps.onClick).toHaveBeenCalled();
  });
});
