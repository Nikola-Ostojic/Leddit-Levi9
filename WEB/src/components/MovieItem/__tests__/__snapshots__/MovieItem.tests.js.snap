// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem component tests Should match snapshot 1`] = `
<WithStyles(ForwardRef(Card))
  className="container"
  onClick={[Function]}
  raised={true}
>
  <WithStyles(ForwardRef(CardContent))>
    Test
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
